Cocktails9/Cocktails9/View/Authentication/LoginView.swift

```diff
@@ -74,28 +81,32 @@ struct LoginView: View {
     
     
     private func login() {
-        if let loadedUser = UserManager.loadUser() {
-            if username == loadedUser.username && password == loadedUser.password {
-                isLoggedIn = true
-                // Successful login, do nothing (just keep going to the next screen or update the UI)
-                alertMessage = ""
-                showAlert = false
-            } else {
-                alertMessage = "Invalid credentials. Please try again."
-                showAlert = true
-            }
+        // Load all users from UserDefaults
+        if let user = UserManager.loadUser(email: email) {
+                if email == user.email && password == user.password {
+                    isLoggedIn = true
+                    alertMessage = ""
+                    showAlert = false
+                    navigateToMain = true
+                    withAnimation {
+                        currentImageName = "cocktailP2"
+                    }
+                    DispatchQueue.main.asyncAfter(deadline: .now() + 1.5) {
+                        navigateToMain = true
+                    }
+                    return
+                }
+            alertMessage = "Invalid credentials. Please try again."
+            showAlert = true
         } else {
-            alertMessage = "No user found. Please register first."
+            alertMessage = "No users found. Please register first."
             showAlert = true
         }
```

Comment:
usual convention is to avoid if else pyramid, by using `guard | else` statements https://docs.swift.org/swift-book/documentation/the-swift-programming-language/statements/#Guard-Statement